examples/gio_dbus_register_object/main.rs

```diff
@@ -94,24 +97,33 @@ mod imp {
             connection
                 .register_object("/com/github/gtk_rs/examples/HelloWorld", &example)
                 .typed_method_call::<HelloMethod>()
-                .invoke_and_return_future_local(|_, sender, call| {
-                    println!("Method call from {sender:?}");
-                    async {
-                        match call {
-                            HelloMethod::Hello(Hello { name }) => {
-                                let greet = format!("Hello {name}!");
-                                println!("{greet}");
-                                Ok(Some(greet.to_variant()))
-                            }
-                            HelloMethod::SlowHello(SlowHello { name, delay }) => {
-                                glib::timeout_future(Duration::from_secs(delay as u64)).await;
-                                let greet = format!("Hello {name} after {delay} seconds!");
-                                println!("{greet}");
-                                Ok(Some(greet.to_variant()))
+                .invoke_and_return_future_local(glib::clone!(
+                    #[strong(rename_to = app)]
```

Comment:
I think this causes a reference cycle. Should probably be a weak reference at least

Reply:
Sure, can change this, though I'm not really following where the cycle would come from, or why it'd be harmful given that the app usually lives for the entire lifetime of the process 🤔

But I'm also only superficially familiar with all this, so I trust your judgement 😅